battlepass/src/lib.rs

```diff
@@ -150,6 +199,56 @@ pub mod pallet {
 		OptionQuery
 	>;
 
+	/// Total earned Points for users per each Battlepass.
+	///
+	/// Points: map (Hash, AccountId) => NftId
+	#[pallet::storage]
+	#[pallet::getter(fn get_points)]
+	pub(super) type Points<T: Config> = StorageDoubleMap<_,
+		Blake2_128Concat, T::Hash,
+		Blake2_128Concat, T::AccountId,
+		u32,
+		ValueQuery
+	>;
+
+	/// Reward by its id.
+	///
+	/// Rewards: map Hash => Reward
+	#[pallet::storage]
+	#[pallet::getter(fn get_reward)]
+	pub(super) type Rewards<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, Reward<T::Hash, String<T>>, OptionQuery>;
+
+	/// Reward state by its id.
+	///
+	/// RewardStates: map Hash => RewardState
+	#[pallet::storage]
+	#[pallet::getter(fn get_reward_state)]
+	pub(super) type RewardStates<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, RewardState, OptionQuery>;
```

Comment:
Why not replace this with `state` field in `Reward` struct?

Reply:
It will not be so efficient if we'll need to update only the state in the struct. See [this](https://youtu.be/9S8rmW8LD5o?t=1059).

Reply:
what does it exactly store? if it is the result of completing a quest, then the possession of the resulting nft is already a proof.

Reply:
Do you mean `RewardStates`? It stores the state of the Reward. Battlepass owner creates `Rewards`. When user claims the Reward it will mint NFT and add a record to the `ClaimedRewards` storage.